billing/views.py

```diff
@@ -130,3 +136,13 @@ def get_object(self):
                 .get(owner=self.request.user)
         except Account.DoesNotExist:
             raise Http404('No Account matches the given query.')
+
+
+@permission_classes([permissions.IsAuthenticated])
+@api_view(['POST'])
+def pay_debt(request):
```

Comment:
you may want to add some preconditions here:
- complain if there is no valid credit card on record
- complain if there are no pending invoices

Reply:
I'm not following you, I don't understand `complain if there is no valid credit card on record`. If user has not a valid card he has not to be marked as complaint if has pending invoices? 
Do you mean these conditions can avoid to call `assign_funds_to_account_pending_invoices` and `charge_pending_invoices`?

Reply:
I mean that if you're trying to somehow guide the user (that is using the javascript frontent) you could tell him when calling this endpoint is not correct (I got this idea from your code that returned a 400).

Reply:
Now I return always a 200 status but with the list of errors (precondition errors)